From f4ef7c5c536f864d6d17b36ee151636ce3a1ab47 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Thu, 6 Apr 2023 14:34:19 +0200
Subject: [PATCH] bytesatwork: bawconfig: introduce Makefile

Use own Makefile to place baw_config beside the board(s).

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit d83cf89382db1dc7f86616cae2d0f66c0ba79100)
---
 board/bytesatwork/baw_config/MAINTAINERS |  1 +
 board/bytesatwork/baw_config/Makefile    | 17 +++++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 board/bytesatwork/baw_config/Makefile

diff --git a/board/bytesatwork/baw_config/MAINTAINERS b/board/bytesatwork/baw_config/MAINTAINERS
index 7c764850c5..ea8cac6787 100644
--- a/board/bytesatwork/baw_config/MAINTAINERS
+++ b/board/bytesatwork/baw_config/MAINTAINERS
@@ -2,3 +2,4 @@ BYTESATWORK BAWCONFIG
 M:	Rico Ganahl <rico.ganahl@bytesatwork.ch>
 S:	Maintained
 F:	board/bytesatwork/baw_config/*
+F:	board/bytesatwork/baw_config/Makefile
diff --git a/board/bytesatwork/baw_config/Makefile b/board/bytesatwork/baw_config/Makefile
new file mode 100644
index 0000000000..67cc78073f
--- /dev/null
+++ b/board/bytesatwork/baw_config/Makefile
@@ -0,0 +1,17 @@
+# Copyright (C) 2023 bytes at work AG
+#
+# SPDX-License-Identifier:      GPL-2.0+
+#
+
+ifdef CONFIG_SPL_BUILD
+obj-y += baw_config.o
+obj-y += baw_config_get.o
+else
+ifneq ($(or $(CONFIG_CMD_BAW_CONFIG), $(CONFIG_BAW_CONFIG_BUILTIN), $(CONFIG_BAW_CONFIG_EEPROM)),)
+obj-y += baw_config.o
+endif
+endif
+
+obj-$(CONFIG_CMD_BAW_CONFIG) += cmd_baw_config.o
+obj-$(CONFIG_BAW_CONFIG_EEPROM) += baw_config_eeprom.o
+obj-$(CONFIG_BAW_CONFIG_BUILTIN) += baw_config_builtin.o
-- 
2.39.2

